### ISSUE TITLE
Setting "Enforce expiration date" prevent any modification of it whatsoever

### ISSUE THREAD

**Issue by AceSlash** [nextcloud/server#15919]
### Steps to reproduce
1. Connect yourself as an admin
2. Go to the sharing settings
3. Check the box "Set default expiration date"
4. Check the box "Enforce expiration date"
5. As a normal user, create a share via a link
6. It is impossible to change the expiration date

### Expected behaviour
In previous versions of Nextcloud, the expiration date could be changed to be less than the maximum expiration date. For example if the expiration date was set to expire after 30 days, it could be changed to expire after 7 days instead.

### Actual behaviour
It is now impossible to change the expiration date, it remains at the value from "Expire after".

### Server configuration

**Operating system**: Ubuntu 18.04

**Web server:** apache httpd

**Database:** postgresql 

**PHP version:** 7.2

**Nextcloud version:** 15.0.8

**Updated from an older Nextcloud/ownCloud or fresh install:** updated since several years

**Where did you install Nextcloud from:** zip from https://download.nextcloud.com/server

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.1.0
  - activity: 2.8.2
  - admin_audit: 1.5.0
  - bookmarks: 1.0.5
  - bruteforcesettings: 1.3.0
  - checksum: 0.4.2
  - cloud_federation_api: 0.1.0
  - comments: 1.5.0
  - dav: 1.8.2
  - federatedfilesharing: 1.5.0
  - federation: 1.5.0
  - files: 1.10.0
  - files_downloadactivity: 1.4.0
  - files_markdown: 2.0.6
  - files_pdfviewer: 1.4.0
  - files_sharing: 1.7.0
  - files_texteditor: 2.7.0
  - files_trashbin: 1.5.0
  - files_versions: 1.8.0
  - files_videoplayer: 1.4.0
  - firstrunwizard: 2.4.0
  - gallery: 18.2.0
  - groupfolders: 3.0.2
  - impersonate: 1.2.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.3.0
  - metadata: 0.9.0
  - nextcloud_announcements: 1.4.0
  - notifications: 2.3.0
  - oauth2: 1.3.0
  - onlyoffice: 2.1.10
  - password_policy: 1.5.0
  - provisioning_api: 1.5.0
  - quota_warning: 1.4.0
  - serverinfo: 1.5.0
  - sharebymail: 1.5.0
  - spreed: 5.0.3
  - support: 1.0.0
  - survey_client: 1.3.0
  - systemtags: 1.5.0
  - theming: 1.6.0
  - twofactor_backupcodes: 1.4.1
  - twofactor_totp: 2.1.2
  - updatenotification: 1.5.0
  - user_ldap: 1.5.0
  - workflowengine: 1.5.0
Disabled:
  - encryption
  - files_external
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

### Client configuration
**Browser:** Firefox 67, Chromium 67

**Comment by ChristophWurst** [comment]
Possible regression from https://github.com/nextcloud/server/commit/260290ec41c61bebb711db174db95fd18ba1aa87#diff-487ee7046537f72c381cd368b403d142R81

**Comment by ChristophWurst** [comment]
cc @skjnldsv

**Comment by skjnldsv** [comment]
Fixed with #15719
